web/router.ex

```diff
@@ -74,7 +74,9 @@ defmodule CoursePlanner.Router do
       put "/update_fill", AttendanceController, :update_fill, as: :update_fill
     end
 
-    resources "/settings", SettingController, only: [:show, :edit, :update], singleton: true
+    resources "/settings", SettingController, only: [:show, :update], singleton: true do
```

Comment:
if you don't change the router you can achieve a similar result using the following links
```elixir
setting_path(@conn, :edit, setting_type: "system") # /settings?setting_type=system
```
or
```elixir
setting_path(@conn, :edit, setting_type: "program") # /settings?setting_type=program
```
the controller can stay the same for this urls

the advantage would be having a standard path helper function